.gitignore

```diff
@@ -1,5 +1,11 @@
+examples/testnet
+decred/dist
+tinywallet/dist
+decred/poetry.lock
```

Comment:
It is, well spotted. `tinywallet/poetry.lock` is indeed committed, `decred/poetry.lock` is not. The idea is to commit lock files for applications, but not for libraries.

You want to commit the lock file for apps so that a build always uses the same versions of the dependencies (and their dependencies), and therefore is reproducible.

You don't want to commit the lock file for libraries because then you would have multiple lock files for different libraries and for the app that uses them, potentially conflicting among them.

The libraries will specify the version intervals for their dependencies in their `pyproject.toml` or `requirements.txt` files; Poetry will respect those intervals, but will disregard any libraries' lock files. The only lock file that counts is the app one.

Reply:
This is probably nothing, but using the provided `tinywallet/poetry.lock` file threw an error at me:
```
[joe@hyrule tinywallet]$ poetry install
Installing dependencies from lock file
Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.

[SolverProblemError]
Because tinywallet depends on pyqt5 (=5.12.3) which doesn't match any versions, version solving failed.
```

Running poetry update solved the problem, but it gave me a rather large diff.

Reply:
Indeed, forgot to update the dependencies and the lock file. Thanks.